client/src/pages/profile/index.tsx

```diff
@@ -0,0 +1,473 @@
+   */
+  const isViewable = (field: keyof IUser): boolean =>
+    permissions.view.includes(field);
+
+  /**
+   * Determines if a field is editable to the current user
+   *
+   * @param field the field to check
+   * @returns true if field is editable, else false
+   */
+  const isEditable = (field: keyof IUser): boolean =>
+    permissions.edit.includes(field) && isEditMode;
+
+  /**
+   * Edits and updates the local user state varialble
+   *
+   * @param field the field of the user to edit
+   * @param value the value corresponding to that field
+   */
+  const editUser = <T extends keyof IUser>(field: T, value: IUser[T]): void =>
+    setUser(updateUserField(user, field, value));
+
+  /**
+   * Gets the cumulative options for a dropdown including
+   * user's current options
+   *
+   * @param dropdown the dropdown to get
+   * @returns the parsed semantic UI dropdown options
+   */
+  const getMultiDropdownOptions = (
+    dropdown: MultiDropdowns,
+  ): DropdownItemProps[] =>
+    parseArrayToSemanticDropdownOptions(
+      dropdownOptions[dropdown].concat(user[dropdown]),
+    );
+
+  /**
+   * Adds an option to a dropdown
+   *
+   * @param dropdown the dropdown to add the option to
+   * @param value the value to add
+   */
+  const addDropdownOption = (dropdown: MultiDropdowns, value: string): void => {
+    let allOptions;
+
+    if (dropdown === 'genders') {
+      allOptions = allGenders;
+    } else if (dropdown === 'pronouns') {
+      allOptions = allPronouns;
+    } else {
+      allOptions = allRaces;
+    }
+
+    updateDropdownOptions(dropdown, value, allOptions);
+  };
+
+  return (
+    <>
+      <Sidebar currentPage={pages.PROFILE} />
+      <Header />
+      <div style={{ marginLeft: '15%', marginTop: '3%' }}>
+        <Grid stackable>
+          <Grid.Row columns={4}>
+            <Grid.Column className="profile-pic-col">
+              {(userId === currentUser._id || currentUser.role === 'ADMIN') && (
+                <Button
+                  size="medium"
+                  className="edit-profile-btn"
+                  onClick={startEditMode}
+                  content="Edit Profile"
+                />
+              )}
+
+              <Image size="small" circular src={getUserProfilePic(user)} />
+
+              {user.masthead && (
+                <Image className="masthead" size="small" src={Masthead} />
+              )}
+
+              {isEditMode && (
+                <Button.Group>
+                  <Button
+                    className="edit-mode-btn save"
+                    content="Save Changes"
+                    onClick={saveUser}
+                    size="medium"
+                  />
+                  <Button
+                    className="edit-mode-btn cancel"
+                    content="Cancel Changes"
+                    onClick={cancelEditMode}
+                    size="medium"
+                  />
+                </Button.Group>
```

Comment:
This section seems to be overlapping into the section next to it. Maybe try messing with the button margins or adding a widths='2' inside the button group.

Reply:
Could you send me a picture of what you are seeing? I can't seem to recreate the overalap

Reply:
![Screen Shot 2021-07-14 at 9 27 43 PM](https://user-images.githubusercontent.com/24413724/125718479-e11f24cc-b87f-462e-9202-266c89f9c827.jpg)

Reply:
I'm guessing it's bc I'm on a smaller screen

Reply:
that is spicy